Library/Homebrew/test/livecheck/strategy/sparkle_spec.rb

```diff
@@ -105,37 +191,89 @@
+    {
+      version_123: Homebrew::Livecheck::Strategy::Sparkle::Item.new(
+        title:          item_hashes[:version_123][:title],
+        pub_date:       Time.parse(item_hashes[:version_123][:pub_date]),
+        url:            item_hashes[:version_123][:url],
+        bundle_version: Homebrew::BundleVersion.new(item_hashes[:version_123][:short_version],
+                                                    item_hashes[:version_123][:version]),
+      ),
+      version_122: Homebrew::Livecheck::Strategy::Sparkle::Item.new(
+        title:          item_hashes[:version_122][:title],
+        # Note `Time.new(0)` rather than
+        # `item_hashes[:version_122][:pub_date]`. This is because when an
+        # unparseable pub_date is present in the XML it nils out that
+        # field in the struct and then substitutes `Time.new(0)` as a
+        # default value.
+        pub_date:       Time.new(0),
+        url:            item_hashes[:version_122][:url],
+        bundle_version: Homebrew::BundleVersion.new(item_hashes[:version_122][:short_version],
+                                                    item_hashes[:version_122][:version]),
+      ),
+      version_121: Homebrew::Livecheck::Strategy::Sparkle::Item.new(
+        title:          item_hashes[:version_121][:title],
+        pub_date:       Time.parse(item_hashes[:version_121][:pub_date]),
+        url:            item_hashes[:version_121][:url],
+        bundle_version: Homebrew::BundleVersion.new(item_hashes[:version_121][:short_version],
+                                                    item_hashes[:version_121][:version]),
+      ),
+      version_120: Homebrew::Livecheck::Strategy::Sparkle::Item.new(
+        title:          item_hashes[:version_120][:title],
+        pub_date:       Time.parse(item_hashes[:version_120][:pub_date]),
+        url:            item_hashes[:version_120][:url],
+        bundle_version: Homebrew::BundleVersion.new(item_hashes[:version_120][:short_version],
+                                                    item_hashes[:version_120][:version]),
+      ),
+    }
+  }
+
   let(:items) {
     items = {
       appcast: [
-        Homebrew::Livecheck::Strategy::Sparkle::Item.new(
-          title:          item_hash[0][:title],
-          pub_date:       Time.parse(item_hash[0][:pub_date]),
-          url:            item_hash[0][:url],
-          bundle_version: Homebrew::BundleVersion.new(item_hash[0][:short_version], item_hash[0][:version]),
-        ),
-        Homebrew::Livecheck::Strategy::Sparkle::Item.new(
-          title:          item_hash[1][:title],
-          pub_date:       Time.new(0),
-          url:            item_hash[1][:url],
-          bundle_version: Homebrew::BundleVersion.new(item_hash[1][:short_version], item_hash[1][:version]),
-        ),
+        version_items[:version_123],
+        version_items[:version_122],
+        version_items[:version_121],
+        version_items[:version_120],
       ],
```

Comment:
```suggestion
      ],
      appcast_sorted: [
        version_items[:version_123],
        version_items[:version_121],
        version_items[:version_120],
        version_items[:version_122],
      ],
```

Instead of creating a sorted array later in the tests, it probably makes more sense to define it here.